### PR TITLE
Add cleanup and do another pass on logging system

### PR DESCRIPTION
**Pull request by fepitre**
We introduce a new cli command that allows to cleanup several files during build process. We also improve again our logging system.

### CODE REVIEW

File: qubesbuilder/cli/cli_cleanup.py [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    obj.dry_run = dry_run
+
+
+def get_component_distfiles(component: QubesComponent):
+    current_distfiles = []
+    try:
+        for file in (
+            component.get_parameters().get("source", {}).get("files", [])
+        ):
+            if "url" in file:
```
Comment (marmarek) [inline qubesbuilder/cli/cli_cleanup.py:35]: `get_archive_name` handles this case too.

File: qubesbuilder/cli/cli_cleanup.py [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+    """
+    Cleanup cache files and directories.
+    """
+    if all:
+        chroot = True
+        installer = True
+
+    to_delete = []
+    if chroot:
+        to_delete.append(obj.config.cache_dir / "chroot")
```
Comment (marmarek) [inline qubesbuilder/cli/cli_cleanup.py:201]: There should be an option to remove chroots not relevant anymore (for no longer enabled dists)

File: qubesbuilder/config.py [hunk 3]
```diff
@@ -365,7 +363,44 @@ def artifacts_dir(self):
...
+    def installer_dir(self):
+        return self.artifacts_dir / "installer"
+
+    @property
+    def iso_dir(self):
+        return self.artifacts_dir / "iso"
+
+    @property
+    def logs_dir(self):
         return self.artifacts_dir / "logs"
```
Comment (marmarek) [inline qubesbuilder/config.py:404]: Old helpers (in the Plugin class) had also `resolve()` call. Is it intentional to skip it here?

**Review by marmarek — CHANGES_REQUESTED** [review]
`gitlab-ci.yml` needs adjustment to run the new tests

File: qubesbuilder/cli/cli_cleanup.py (hunk 2 above)
Comment (fepitre) [reply qubesbuilder/cli/cli_cleanup.py:201]: Right now it's going to remove all chroots, is it not what's wanted?

File: qubesbuilder/config.py (hunk 3 above)
Comment (fepitre) [reply qubesbuilder/config.py:404]: Yes, self.artifacts_dir is resolved already.

File: qubesbuilder/cli/cli_cleanup.py (hunk 2 above)
Comment (fepitre) [reply qubesbuilder/cli/cli_cleanup.py:199]: I've added an option to only delete unused chroot based on distributions provided in builder.yml

**Comment by marmarek** [comment]
I'm going to add trivial `get_logger()` function to keep API compatibility

**Review by marmarek — APPROVED** [review]

**Pull request merged**